### PR TITLE
feature/wallet-model

### PR DESCRIPTION
**Pull request by Jifflis**
# 📲 What #
- Create Imageable model
 
# 🛠 How #
- added Wallet model class and  Wallet migration class
 
# ✅ Acceptance criteria #

# ⏰ TODO #
None

# 🔗 References #
https://trello.com/c/nl4MiP6M/5-wallet-model

### CODE REVIEW

**Comment by jaimejazarenoiii** [comment]
Use PR tempalate

File: app/models/wallet.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#  currency    :integer
+#  target_date :datetime
+#  title       :string
+#  total       :float
+#  type        :string
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+class Wallet < ApplicationRecord
+	enum type: %i[type1 type2]
```
Comment (jaimejazarenoiii) [inline app/models/wallet.rb]: missing enum for currencies

File: db/schema.rb [hunk 2]
```diff
@@ -12,24 +10,44 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20_210_323_160_745) do
+ActiveRecord::Schema.define(version: 2021_03_31_054326) do
+
   # These are extensions that must be enabled in order to support this database
-  enable_extension 'plpgsql'
+  enable_extension "plpgsql"
```
Comment (jaimejazarenoiii) [inline db/schema.rb]: again, rubocop will mark this as an offense, always make sure, overcommit is not bypassed 
Screen Shot 2021-04-01 at 10 17 16 AM

**Review by jaimejazarenoiii — CHANGES_REQUESTED** [review]

File: db/migrate/20210331054326_create_wallets.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CreateWallets < ActiveRecord::Migration[6.1]
+  def change
+    create_table :wallets do |t|
+      t.string :title
```
Comment (jaimejazarenoiii) [inline db/migrate/20210331054326_create_wallets.rb]: missing extra properties for checking if this one is nullable

File: test/models/wallet_test.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+#  title       :string           not null
+#  total       :float
+#  type        :string           not null
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+require 'test_helper'
+
+class WalletTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (jaimejazarenoiii) [inline test/models/wallet_test.rb]: Please add tests

File: app/models/wallet.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+#  title       :string           not null
+#  total       :float
+#  type        :string           not null
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+class Wallet < ApplicationRecord
+  enum currency: %i[php usd]
+  validates :title, presence: true, uniqueness: true
+  validates :type, :currency, presence: true
```
Comment (jaimejazarenoiii) [inline app/models/wallet.rb]: I think type should be an enum also
normal and goal are the values

**Review by rexendy — APPROVED** [review]

**Pull request merged**